### PR TITLE
Update to RN 0.73

### PR DESCRIPTION
**Pull request by acoates-ms**
Mostly package version bumps.

In 0.72 react-native-windows was not exporting its type info correctly, so the window type adapters were just using RN's types.  Now they are more complete. With 0.73, react-native-windows is more correctly exporting its TS types, which requires updates to our adapter types.  Now that we have both windows and win32 types in the adapters, it more clearly shows some of the inconsistencies between the two platforms which we should work to reduce.

### CODE REVIEW

**Comment by Saadnajmi** [comment]
This will be blocked by https://github.com/microsoft/fluentui-react-native/pull/3376 as an FYI

File: packages/dependency-profiles/src/index.js [hunk 1]
```diff
@@ -1,636 +1,946 @@
...
-    "@fluentui-react-native/tokens": {
-      "name": "@fluentui-react-native/tokens",
-      "version": "0.22.9"
-    }
   },
-  "0.71": {
-    "@fluentui-react-native/e2e-testing": {
-      "name": "@fluentui-react-native/e2e-testing",
-      "version": "1.32.8"
+  0.72: {
```
Comment (rurikoaraki) [inline packages/dependency-profiles/src/index.js:314]: We have an existing 0.71 file I think? Do we need one for 0.72?

**Comment by rurikoaraki** [comment]
FYI #3544 added a few more packages to dep profiles.

**Comment by Saadnajmi** [comment]
For the macOS error, looks like codegen doesn't like the object `ImageSourcePropType` for MenuButton's native component. As a fix, we could probably just remove the native module altogether and use the JS version (which itself uses a Callout that _is_ implemented on macOS)

File: scripts/src/configs/configureJest.ts [hunk 2]
```diff
@@ -18,6 +18,9 @@ export function configureJest(customConfig?: object): object {
       moduleNameMapper: {
         '\\.(scss)$': path.resolve(__dirname, 'jest/jest-style-mock.js'),
         KeyCodes: path.resolve(__dirname, 'jest/jest-mock.js'),
+        // Jest is picking up the hoisted version of lru-cache, which is
+        // incompatible from the version required by semver
+        'lru-cache': require.resolve('lru-cache', {paths:[require.resolve('semver')]}),
```
Comment (Saadnajmi) [inline scripts/src/configs/configureJest.ts:23]: Could we fix this with a resolution? Or do we just _need_ to have two copies?

Comment (Saadnajmi) [reply scripts/src/configs/configureJest.ts:23]: Actually, do we even need semver? I see it in our package.json but not actually imported in any JS

**Comment by Saadnajmi** [comment]
The Android issue might be solved by updating the Android test app. Whenever RNTA makes updates to the template files, we don't automatically get them, unless we run `yarn init-test-app` again. I know there was a bunch of Android fixes recently.

**Review by rurikoaraki — APPROVED** [review]

File: .ado/templates/setup-repo.yml [hunk 3]
```diff
@@ -1,9 +1,9 @@
 # This template install dependencies, builds, and bundles this repo
 steps:
-  - task: NodeTool@0
-    inputs:
-      versionSource: 'fromFile'
-      versionFilePath: '.node-version'
+  # - task: NodeTool@0
```
Comment (rurikoaraki) [inline .ado/templates/setup-repo.yml:3]: should this stay commented?

Comment (acoates-ms) [reply .ado/templates/setup-repo.yml:3]: The windows build doesn't work if we use NodeTool to install node (even though its already got node 18?  🤷.  -- And all the builds are working currently.  I can remove the code completely as a follow up PR.

**Pull request merged**